allennlp/models/reading_comprehension/dialog_qa.py

```diff
@@ -0,0 +1,407 @@
+            output_dict['yesno'].append(per_dialog_yesno_list)
+            output_dict['followup'].append(per_dialog_followup_list)
+        return output_dict
+
+    @overrides
+    def decode(self, output_dict: Dict[str, torch.Tensor]) -> Dict[str, Any]:
+        yesno_tags = [[self.vocab.get_token_from_index(x, namespace="yesno_labels") for x in yn_list] \
+                      for yn_list in output_dict.pop("yesno")]
+        followup_tags = [[self.vocab.get_token_from_index(x, namespace="followup_labels") for x in followup_list] \
+                         for followup_list in output_dict.pop("followup")]
+        output_dict['yesno'] = yesno_tags
+        output_dict['followup'] = followup_tags
+        return output_dict
+
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+        return {'start_acc': self._span_start_accuracy.get_metric(reset),
+                'end_acc': self._span_end_accuracy.get_metric(reset),
+                'span_acc': self._span_accuracy.get_metric(reset),
+                'yesno': self._span_yesno_accuracy.get_metric(reset),
+                'followup': self._span_followup_accuracy.get_metric(reset),
+                'f1': self._official_f1.get_metric(reset), }
+
+    @staticmethod
+    def _get_best_span_yesno_followup(span_start_logits: torch.Tensor,
+                                      span_end_logits: torch.Tensor,
+                                      span_yesno_logits: torch.Tensor,
+                                      span_followup_logits: torch.Tensor) -> torch.Tensor:
+        # Returns the index of highest-scoring span that is not longer than 30 tokens, as well as
+        # yesno prediction bit and followup prediction bit from the predicted span end token.
+        if span_start_logits.dim() != 2 or span_end_logits.dim() != 2:
+            raise ValueError("Input shapes must be (batch_size, passage_length)")
+        batch_size, passage_length = span_start_logits.size()
+        max_span_log_prob = [-1e20] * batch_size
+        span_start_argmax = [0] * batch_size
+
+        best_word_span = span_start_logits.new_zeros((batch_size, 4), dtype=torch.long)
+
+        span_start_logits = span_start_logits.data.cpu().numpy()
+        span_end_logits = span_end_logits.data.cpu().numpy()
+        span_yesno_logits = span_yesno_logits.data.cpu().numpy()
+        span_followup_logits = span_followup_logits.data.cpu().numpy()
+        for b_i in range(batch_size):  # pylint: disable=invalid-name
+            for j in range(passage_length):
+                val1 = span_start_logits[b_i, span_start_argmax[b_i]]
+                if val1 < span_start_logits[b_i, j]:
+                    span_start_argmax[b_i] = j
+                    val1 = span_start_logits[b_i, j]
+                val2 = span_end_logits[b_i, j]
+                if val1 + val2 > max_span_log_prob[b_i]:
+                    if j - span_start_argmax[b_i] > 30:
```

Comment:
It's better to have magic numbers like this specified by a variable in `__init__`.  So you could have something like `self._max_span_length = 30`, and use `self._max_span_length` here.